src/mrinufft/trajectories/display.py

```diff
@@ -1,54 +1,121 @@
 """Display function for trajectories."""
+import itertools
+
+import matplotlib as mpl
 import matplotlib.pyplot as plt
 import matplotlib.ticker as mticker
 import numpy as np
 
 from .utils import (
-    compute_gradients_and_slew_rates,
-    convert_trajectory_to_gradients,
-    KMAX,
     DEFAULT_GMAX,
-    DEFAULT_SMAX,
     DEFAULT_RASTER_TIME,
+    DEFAULT_SMAX,
+    KMAX,
+    compute_gradients_and_slew_rates,
+    convert_trajectory_to_gradients,
 )
 
 
-COLOR_CYCLE = [
-    "tab:blue",
-    "tab:orange",
-    "tab:green",
-    "tab:red",
-    "tab:purple",
-    "tab:olive",
-    "tab:pink",
-    "tab:cyan",
-]
-NB_COLORS = len(COLOR_CYCLE)
-
-
 class displayConfig:
     """
     A container class used to share arguments related to display.
 
-    Attributes
-    ----------
-    alpha : float
-        Transparency used for area plots, by default 0.2.
-    linewidth : float
-        Width for lines or curves, by default 2.
-    pointsize : float
-        Size for points used to show constraints, by default 10.
-    fontsize : float
-        Font size for most labels and texts, by default 18.
-    small_fontsize : float
-        Font size for smaller texts, by default 14.
+    The values can be updated either directy (and permanently) or temporarily by using
+    a context manager.
+
+    Examples
+    --------
+    >>> from mrinufft.trajectories.display import displayConfig
+    >>> displayConfig.alpha
+    0.2
+    >>> with displayConfig(alpha=0.5):
+            print(displayConfig.alpha)
+    0.5
+    >>> displayConfig.alpha
+    0.2
     """
 
-    alpha = 0.2
-    linewidth = 2
-    pointsize = 10
-    fontsize = 18
-    small_fontsize = 14
+    alpha: float = 0.2
+    """Transparency used for area plots, by default ``0.2``."""
+    linewidth: float = 2
+    """Width for lines or curves, by default ``2``."""
```

Comment:
Is it conventional to comment after the variable definition ?

Reply:
Yes ! It avoids having an "Attributes" section in the docstring. Sphinx then produce a nice looking summary.